roles/postfix/tasks/main.yml

```diff
@@ -0,0 +1,58 @@
+---
+# tasks file for roles/postfix
+- name: Postfix | Install Postfix
+  ansible.builtin.apt:
+    name: postfix
+    state: present
+    update_cache: true
+
+- name: Postfix | Configure Postfix as relay host
+  ansible.builtin.lineinfile:
+    path: "{{ postfix_main_cf }}"
+    line: "{{ item }}"
+    state: present
+  loop:
+    - "relayhost = [{{ smtp_relay_host }}]:{{ smtp_relay_port }}"
+    - "mynetworks = /etc/postfix/mynetworks"
+    - "smtp_use_tls = yes"
+    - "smtp_tls_security_level = may"
+    - "smtp_sasl_auth_enable = yes"
+    - "smtp_sasl_security_options ="
+    - "smtp_sasl_tls_security_options ="
+    - "smtp_sasl_password_maps = hash:/etc/postfix/sasl_passwd"
+    - "smtp_tls_CAfile = /etc/ssl/certs/ca-certificates.crt"
+
+- name: Postfix | Add Proofpoint authentication credentials
+  ansible.builtin.copy:
+    dest: /etc/postfix/sasl_passwd
+    content: |
+      [{{ smtp_relay_host }}]:{{ smtp_relay_port }} {{ smtp_relay_username }}:{{ smtp_relay_password }}
+    mode: '0644'
+  when: running_on_server
+  notify: restart postfix
+
+- name: Postfix | Secure the password file
+  ansible.builtin.file:
+    path: /etc/postfix/sasl_passwd
+    owner: root
+    group: root
+    mode: '0600'
+  when: running_on_server
+
+- name: Postfix | Generate Postfix hash map for authentication
+  ansible.builtin.command: postmap /etc/postfix/sasl_passwd
+  changed_when: false
+  when: running_on_server
+  notify: restart postfix
+
+- name: Postfix | Configure Postfix relay restrictions
+  ansible.builtin.lineinfile:
```

Comment:
Using `lineinfile` means we never clean out old IP addresses. This could come back to bite us if we re-use an old IP for something that should not have postfix functionality.

Reply:
opted to use templates instead. Thanks